Compilation error with kernel >=4.11.0

Build error when building against linux-4.11.0:

```
*  CC [M]  /var/tmp/portage/dev-util/sysdig-0.15.1/work/sysdig-0.15.1/driver/ppm_events.o
*  CC [M]  /var/tmp/portage/dev-util/sysdig-0.15.1/work/sysdig-0.15.1/driver/ppm_fillers.o
*  CC [M]  /var/tmp/portage/dev-util/sysdig-0.15.1/work/sysdig-0.15.1/driver/event_table.o
*  CC [M]  /var/tmp/portage/dev-util/sysdig-0.15.1/work/sysdig-0.15.1/driver/syscall_table.o
*  CC [M]  /var/tmp/portage/dev-util/sysdig-0.15.1/work/sysdig-0.15.1/driver/ppm_cputime.o
*/var/tmp/portage/dev-util/sysdig-0.15.1/work/sysdig-0.15.1/driver/main.c:122:55: error: unknown type name ‘cputime_t’
* void ppm_task_cputime_adjusted(struct task_struct *p, cputime_t *ut, cputime_t
*                                                       ^
*/var/tmp/portage/dev-util/sysdig-0.15.1/work/sysdig-0.15.1/driver/main.c:122:70: error: unknown type name ‘cputime_t’
* oid ppm_task_cputime_adjusted(struct task_struct *p, cputime_t *ut, cputime_t *
*                                                                     ^
*/var/tmp/portage/dev-util/sysdig-0.15.1/work/sysdig-0.15.1/driver/main.c: In function ‘ppm_ioctl’:
*/var/tmp/portage/dev-util/sysdig-0.15.1/work/sysdig-0.15.1/driver/main.c:620:3: error: implicit declaration of function
 ‘for_each_process’ [-Werror=implicit-function-declaration]
*   for_each_process(p) {
*   ^
*/var/tmp/portage/dev-util/sysdig-0.15.1/work/sysdig-0.15.1/driver/main.c:620:23: error: expected ‘;’ before ‘{’ token
*   for_each_process(p) {
*                       ^
*/var/tmp/portage/dev-util/sysdig-0.15.1/work/sysdig-0.15.1/driver/main.c:591:27: warning: unused variable ‘t’ [-Wunused
-variable]
*   struct task_struct *p, *t;
*                           ^
*cc1: some warnings being treated as errors

```

--- a/driver/main.c
+++ b/driver/main.c
@@ -38,7 +38,12 @@ along with sysdig.  If not, see <http://www.gnu.org/licenses/>.
 #include <linux/kdev_t.h>
 #include <linux/delay.h>
 #include <linux/proc_fs.h>
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 11, 0))
 #include <linux/sched.h>
+#else
+#include <linux/sched/signal.h>
+#include <linux/sched/cputime.h>
+#endif
 #include <linux/vmalloc.h>
 #include <linux/wait.h>
 #include <linux/tracepoint.h>
@@ -119,7 +124,9 @@ static void record_event_all_consumers(enum ppm_event_type event_type,
 static int init_ring_buffer(struct ppm_ring_buffer_context *ring);
 static void free_ring_buffer(struct ppm_ring_buffer_context *ring);
 static void reset_ring_buffer(struct ppm_ring_buffer_context *ring);
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 4, 0))
 void ppm_task_cputime_adjusted(struct task_struct *p, cputime_t *ut, cputime_t *st);
+#endif
 
 #ifndef CONFIG_HAVE_SYSCALL_TRACEPOINTS
  #error The kernel must have HAVE_SYSCALL_TRACEPOINTS in order for sysdig to be useful
@@ -624,7 +631,11 @@ static long ppm_ioctl(struct file *filp, unsigned int cmd, unsigned long arg)
 				task_lock(p);
 #endif
 				if (nentries < pli.max_entries) {
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 11, 0))
 					cputime_t utime, stime;
+#else
+					u64 utime, stime;
+#endif
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 4, 0))
 					task_cputime_adjusted(t, &utime, &stime);
@@ -632,8 +643,13 @@ static long ppm_ioctl(struct file *filp, unsigned int cmd, unsigned long arg)
 					ppm_task_cputime_adjusted(t, &utime, &stime);
 #endif
 					proclist_info->entries[nentries].pid = t->pid;
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 11, 0))
 					proclist_info->entries[nentries].utime = cputime_to_clock_t(utime);
 					proclist_info->entries[nentries].stime = cputime_to_clock_t(stime);
+#else
+					proclist_info->entries[nentries].utime = utime;
+					proclist_info->entries[nentries].stime = stime;
+#endif
 				}
 
 				nentries++;

--- a/driver/main.c
+++ b/driver/main.c
@@ -647,8 +647,8 @@ static long ppm_ioctl(struct file *filp, unsigned int cmd, unsigned long arg)
 					proclist_info->entries[nentries].utime = cputime_to_clock_t(utime);
 					proclist_info->entries[nentries].stime = cputime_to_clock_t(stime);
 #else
-					proclist_info->entries[nentries].utime = utime;
-					proclist_info->entries[nentries].stime = stime;
+					proclist_info->entries[nentries].utime = nsec_to_clock_t(utime);
+					proclist_info->entries[nentries].stime = nsec_to_clock_t(stime);
 #endif
 				}
 

--- a/driver/ppm_cputime.c
+++ b/driver/ppm_cputime.c
@@ -321,3 +321,27 @@ void ppm_task_cputime_adjusted(struct task_struct *p, cputime_t *ut, cputime_t *
 #endif /* (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 8, 0)) */
 #endif /* (defined CONFIG_VIRT_CPU_ACCOUNTING_NATIVE) || (LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 30)) */
 #endif /* (LINUX_VERSION_CODE < KERNEL_VERSION(4, 4, 0)) */
+
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0))
+#include <linux/time.h>
+#include <asm-generic/param.h>
+
+/*
+ * Implementation copied from kernel/time/time.c in 4.11.0
+ */
+u64 nsec_to_clock_t(u64 x)
+{
+#if (NSEC_PER_SEC % USER_HZ) == 0
+	return div_u64(x, NSEC_PER_SEC / USER_HZ);
+#elif (USER_HZ % 512) == 0
+	return div_u64(x * USER_HZ / 512, NSEC_PER_SEC / 512);
+#else
+	/*
+         * max relative error 5.7e-8 (1.8s per year) for USER_HZ <= 1024,
+         * overflow after 64.99 years.
+         * exact for HZ=60, 72, 90, 120, 144, 180, 300, 600, 900, ...
+         */
+	return div_u64(x * 9, (9ull * NSEC_PER_SEC + (USER_HZ / 2)) / USER_HZ);
+#endif
+}
+#endif /* (LINUX_VERSION_CODE < KERNEL_VERSION(4, 11, 0)) */

--- a/driver/ppm_cputime.c
+++ b/driver/ppm_cputime.c
@@ -324,7 +324,7 @@ void ppm_task_cputime_adjusted(struct task_struct *p, cputime_t *ut, cputime_t *
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0))
 #include <linux/time.h>
-#include <asm-generic/param.h>
+#include <linux/param.h>
 
 /*
  * Implementation copied from kernel/time/time.c in 4.11.0
@@ -337,10 +337,10 @@ u64 nsec_to_clock_t(u64 x)
 	return div_u64(x * USER_HZ / 512, NSEC_PER_SEC / 512);
 #else
 	/*
-         * max relative error 5.7e-8 (1.8s per year) for USER_HZ <= 1024,
-         * overflow after 64.99 years.
-         * exact for HZ=60, 72, 90, 120, 144, 180, 300, 600, 900, ...
-         */
+	 * max relative error 5.7e-8 (1.8s per year) for USER_HZ <= 1024,
+	 * overflow after 64.99 years
+	 * exact for HZ=60, 72, 90, 120, 144, 180, 300, 600, 900, ...
+	 */
 	return div_u64(x * 9, (9ull * NSEC_PER_SEC + (USER_HZ / 2)) / USER_HZ);
 #endif
 }

